### PR TITLE
gh-353  refactored address text

### PR DESCRIPTION
**Pull request by DmitryBespalov**
Created SlicedText instead of the AddressText so that the whole thing is extendable and composable

### CODE REVIEW

File: Multisig.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3332,6 +3356,7 @@
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
 				ENABLE_NS_ASSERTIONS = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				ENABLE_TESTABILITY = YES;
```
Comment (DmitryBespalov) [inline Multisig.xcodeproj/project.pbxproj:3355]: this was enabled when I clicked "fix warning"

File: Multisig/UI/UI Library/Address/SlicedText+Address.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//  Created by Dmitry Bespalov on 15.07.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import SwiftUI
+
+extension SlicedText.Style {
+    static let addressShortLight = Self.init(color: .gnoMediumGrey, truncate: .middle)
+    static let addressShortDark = Self.init(color: .gnoDarkBlue, truncate: .middle)
+    static let addressLong = Self.init(middleColor: .gnoMediumGrey, sideColor: .gnoDarkBlue, truncate: .none)
```
Comment (DmitryBespalov) [inline Multisig/UI/UI Library/Address/SlicedText+Address.swift:14]: @MouazAlzahabi now the styling is customizable - just add or remove or change in one place.

File: Multisig/UI/UI Library/Address/AddressCell.swift [hunk 3]
```diff
@@ -32,36 +32,32 @@ struct AddressCell: View {
...
                 Spacer()
-
                 BrowseAddressView(address: address)
             }
         }
         .padding(EdgeInsets(top: 2, leading: 0, bottom: 6, trailing: 0))
         .frame(height: 50)
     }
 
-    var addressText: some View {
```
Comment (DmitryBespalov) [inline Multisig/UI/UI Library/Address/AddressCell.swift:49]: this file will be the next target for the refactoring

File: Multisig/UI/Transactions/TransactionCell/SettingsChangeTransactionCell/SetFallbackHandlerSummaryView.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+//  SetFallbackHandlerSummaryView.swift
+//  Multisig
+//
+//  Created by Dmitry Bespalov on 15.07.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import SwiftUI
+
+struct SetFallbackHandlerSummaryView: View {
```
Comment (DmitryBespalov) [inline Multisig/UI/Transactions/TransactionCell/SettingsChangeTransactionCell/SetFallbackHandlerSummaryView.swift]: oops, this needs to be deleted

File: Multisig/UI/UI Library/Address/SlicedText+Address.swift (hunk 2 above)
Comment (MouazAlzahabi) [reply Multisig/UI/UI Library/Address/SlicedText+Address.swift:14]: really nice :)

File: Multisig/UI/UI Library/Text/SlicedString.swift [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+//  Multisig
+//
+//  Created by Dmitry Bespalov on 15.07.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+
+struct SlicedString {
+
```
Comment (MouazAlzahabi) [inline Multisig/UI/UI Library/Text/SlicedString.swift]: Do we need extra lines here

**Review by MouazAlzahabi — APPROVED** [review]

**Review by sche — APPROVED** [review]

**Pull request merged**